### DISCUSSION TITLE
"Best" Flight Controller for INav? Quad

### DISCUSSION THREAD

**Discussion by Tjjohnston** [iNavFlight/inav#8422] [Q&A]
I am in the market to build a new drone, a cinelifter type, trying to make it as small as possible. It's a unique one for sure. 
I guess I have two parts to this. First, is there any quadcopter FC that has Servo support for a gimbal? One that can be controlled from the transmitter. 
If that is not a possibility, then I would ask what is the best overall flight controller as far as compatibility, or even which ones get updated first, that kind of thing. 

Some features I am looking to add:
Lighting (high lumen)
Lidar (matek board)
GPS - BN880
I am also really interested in having the camera switch board, to have a few different camera angles. 

In the past I have bought electronics and then find out they are not compatible with INav, I am  still learning and making a lot of mistakes so I thought I would ask you guys first this time. 

Thank you!

**Comment by EhAye** [comment]
determine how many uarts and what features you need to run your stuff, 

then find FC's that match those specifications, and search that model in the "issues" section in here to see what problems people have with that particular flight controller

if you see a lot of people having issues, maybe avoid that FC. And personally something that's worked out for me really well so far, is picking 8S capable hardware for my 6S setups, and 6S hardware for my 4S stuff. No blown FETs and stuff yet.

And make sure you get a GPS module with built-in compass if you're putting it on a quad. inav needs compass for quads. Which means your FC will need the two extra pads for that
